### PR TITLE
Scav rifle

### PR DESCRIPTION
**Pull request by Baldir-Odensen**
## About The Pull Request

Adds the Scav rifle so that those who wich to make scav themed ruins and to when the scav ship is added give them a gun to use

## Why It's Good For The Game

gives the scavengers a unique gun that is slightly worse than most weapons so that they can have and buy guns without unbalancing the game by making it easy for them to get better weapons than station crew.

## Changelog
:cl:
add: added the scav rifle
/:cl:

### CODE REVIEW

**Comment by zeskorion** [comment]
why do they even need guns at all?

**Comment by Baldir-Odensen** [comment]
It's to be able to figth the hostile mobs in space. It will also be used for the scav contractor simple mob that might be put into new space ruins.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -344,3 +344,12 @@
 	actions_types = list()
 	fire_sound = 'sound/weapons/laser.ogg'
 	casing_ejector = FALSE
+/obj/item/gun/ballistic/automatic/scav
```
Comment (ike709) [inline code/modules/projectiles/guns/ballistic/automatic.dm]: Needs a line of whitespace before it.

**Comment by Baldir-Odensen** [comment]
Ok I'll do that

**Comment by ike709** [comment]
Honestly this should just be a surplus rifle subtype. Or just, y'know, use surplus rifles.

**Comment by Baldir-Odensen** [comment]
The surplus rifle is useless. It's supposed to get its own sprite but I can't sprite so I'll let people who know how to do it do it later. Also it has no stat in common with the surplus rifle except sprite so the amount of code would be the same

**Comment by ike709** [comment]
Actually, everything should match the surplus rifle except the name, desc, fire_delay, and mag_type. Your weapon is just missing various flags that it should have, which the surplus rifle does. More of its vars should be matching the surplus rifle than not, hence why it should be a subtype.

**Comment by Baldir-Odensen** [comment]
It's not supposed to have things like heavy and huge. It's supposed to be something like a pipe smg (sten gun, grease gun) so something smallish not huge and bulky. It's supposed to be like the sec auto rifle but slightly worse.

File: code/modules/projectiles/boxes_magazines/external/rifle.dm [hunk 2]
```diff
@@ -1,22 +1,22 @@
-/obj/item/ammo_box/magazine/m10mm/rifle
```
Comment (qwertyquerty) [inline code/modules/projectiles/boxes_magazines/external/rifle.dm:1]: You changed the line endings in this file. Fix.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 3]
```diff
@@ -344,3 +344,13 @@
 	actions_types = list()
 	fire_sound = 'sound/weapons/laser.ogg'
 	casing_ejector = FALSE
+
+/obj/item/gun/ballistic/automatic/proto/unrestricted/scav
+	name = "Scav gun"
```
Comment (qwertyquerty) [inline code/modules/projectiles/guns/ballistic/automatic.dm:349]: Lowercase scav

**Review by qwertyquerty — CHANGES_REQUESTED** [review]

**Comment by qwertyquerty** [comment]
@Baldir-Odensen ?

**Comment by qwertyquerty** [comment]
@Baldir-Odensen

**Comment by qwertyquerty** [comment]
stale

**Pull request closed without merging**